app/routes/_landing.($lang).docs/route.tsx

```diff
@@ -30,31 +30,63 @@ export const loader: LoaderFunction = async ({ params }) => {
   }
 }
 
-const replaceAll = (str: string, find: string, replace: string) => {
-  return str.replace(new RegExp(find, "g"), replace)
-}
-
 export default function DocsLayout() {
   const { data }: LoaderData = useLoaderData()
   const [linksGroupItems, setLinksGroupItems] = useState<LinksGroupProps[]>([])
 
   useEffect(() => {
     if (data && data.length) {
-      setLinksGroupItems(
-        data.map((doc: documentation) => {
-          return {
-            label:
-              doc.translations && doc.translations[0]?.title
-                ? doc.translations[0].title
-                : "",
-            link: doc.slug ? doc.slug : "",
-            slug: doc.slug ? doc.slug : "",
-          }
-        }),
-      )
+      const organizedData = organizeData(data)
+      setLinksGroupItems(organizedData)
     }
   }, [data])
 
+  const findNestingLevel = (doc: documentation): number => {
+    if (!doc.parent) {
+      return 0
+    }
+    return 1 + findNestingLevel(doc.parent)
+  }
+
+  const organizeData = (docs: documentation[]): LinksGroupProps[] => {
+    let mappedDocs: LinksGroupProps[] = []
+
+    docs.forEach((doc) => {
```

Comment:
imo it's better to use `map` here and avoid pushing. Also, for the usage of the variable below, instead of replacing the array with a new version of itself using map we can create a new const and return it. 
The code this way stays more readable and easy to understand.